src/js/08-theme.js

```diff
@@ -1,31 +1,43 @@
 /* global localStorage */
 ;(function () {
   'use strict'
+  const stored = sessionStorage.getItem('theme')
+  console.log("stored = " + stored)
   const systemInitiatedDark = window.matchMedia('(prefers-color-scheme: dark)')
-  if (systemInitiatedDark.matches) {
+  console.log("system = " + (systemInitiatedDark.matches ? "dark" : "light"))
+  var actual = stored
+  if (actual == null) {
+    actual = systemInitiatedDark.matches ? "dark" : "light"
+    console.log("using system")
+  }
+  console.log("actual = " + actual)
+
+  if (actual === "dark") {
     document.documentElement.setAttribute('data-theme', 'dark')
+    console.log("set element dark")
+  } else {
+    document.documentElement.removeAttribute('data-theme')
+    console.log("set element light")
   }
-  document.addEventListener('DOMContentLoaded', function () {
-    if (systemInitiatedDark.matches) {
-      document.documentElement.setAttribute('data-theme', 'dark')
-    }
-  })
+
+  // document.addEventListener('DOMContentLoaded', function () {
+  //   if (actual === "dark") {
+  //     document.documentElement.setAttribute('data-theme', 'dark')
+  //   } else {
+  //     document.documentElement.removeAttribute('data-theme')
+  //   }
+  // })
```

Comment:
idk what this callback does; seems unnecessary

Reply:
I think it was to make absolute sure once the dom loaded that the correct theme is in place, but if it's working without it, that's cool 😄

Reply:
ohh now I remember, it was because originally I had the attribute set on the body tag and the dom needed to be loaded to add it there, then later I changed it to the html tag instead since that's always loaded, but never removed the listener for it 😭